pkg/did/creator/creator.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright Avast Software. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package creator defines a type that can be used to create DIDs using various supported DID methods.
+package creator
+
+import (
+	"fmt"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+
+	"github.com/trustbloc/wallet-sdk/pkg/api"
+	"github.com/trustbloc/wallet-sdk/pkg/did/key"
+)
+
+// DIDCreator is used for creating DID Documents using supported DID methods (currently only did:key).
+type DIDCreator struct{}
+
+// NewDIDCreator returns a new DIDCreator.
+func NewDIDCreator() *DIDCreator {
```

Comment:
Pass in KeyHandleWriter (create key pair if none specified) and KeyHandleReader (both optional)